zerver/views/streams.py

```diff
@@ -258,7 +258,9 @@ def add_subscriptions_backend(request, user_profile,
                               authorization_errors_fatal = REQ(validator=check_bool, default=True)):
 
     if not user_profile.can_create_streams():
-        return json_error('User cannot create streams.')
+        for stream in streams_raw:
+            if get_stream(stream["name"], user_profile.realm) is None:
+                return json_error('User cannot create streams.')
```

Comment:
Without this change if a user is not allowed to create new streams, he will also not be able to subscribe to existing ones.
The only downside right now is that if a user tries to create a stream and subscribe to others with a single request he will not subscribe to any streams.